lib/test/reporter/dot.js

```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+module.exports = async function*(source) {
+  for await (const { type } of source) {
+    if (type === 'test:fail' || type === 'test:pass') {
```

Comment:
I would consider having a separate `test` event (either to replace these two events or in addition to them). Otherwise reporters need to potentially keep two event streams in sync.